apps/open-swe/langbench/utils.ts

```diff
@@ -0,0 +1,227 @@
+import { createLogger, LogLevel } from "../src/utils/logger.js";
+import { ENV_CONSTANTS } from "../src/utils/env-setup.js";
+import { TestResults, PytestJsonReport, RunPytestOptions } from "./types.js";
+import { readFile } from "../src/utils/read-write.js";
+
+const logger = createLogger(LogLevel.DEBUG, "Langbench Utils");
+
+/**
+ * Fetch diff content from a diff URL and extract test file names, this function is used in one-off situtations to get the test files from the diff url.
+ *
+ */
+export async function getTestFilesFromDiff(diffUrl: string): Promise<string[]> {
+  try {
+    const response = await fetch(diffUrl);
+    if (!response.ok) {
+      throw new Error(`Failed to fetch diff: ${response.statusText}`);
+    }
+
+    const diffContent = await response.text();
+    const testFiles: string[] = [];
+
+    // Parse the diff to find modified files
+    const lines = diffContent.split("\n");
+    for (const line of lines) {
+      // Look for diff file headers
+      if (line.startsWith("diff --git ")) {
+        const match = line.match(/diff --git a\/(.+?) b\//);
+        if (match) {
+          const filePath = match[1];
+          // Check if this is a test file in libs/langgraph/tests/
+          if (isLangGraphTestFile(filePath)) {
+            testFiles.push(filePath);
+          }
+        }
+      }
+    }
+
+    return [...new Set(testFiles)]; // Remove duplicates
+  } catch (error) {
+    logger.error(`Failed to fetch or parse diff from ${diffUrl}:`, { error });
+    return [];
+  }
+}
+
+/**
+ * Check if a file path represents a test file in libs/langgraph/tests/
+ */
+export function isLangGraphTestFile(filePath: string): boolean {
```

Comment:
nit, drop export if only used in this file
```suggestion
function isLangGraphTestFile(filePath: string): boolean {
```